lib/spack/spack/cmd/ci.py

```diff
@@ -413,20 +413,19 @@ def ci_rebuild(args):
                 # install from buildcache even though the locally computed
                 # full hash is different than the one stored in the spec.yaml
                 # file on the remote mirror.
-                if job_spec.dependencies():
```

Comment:
@opadron This is the guard I was mentioning a bit ago.  Removing it doesn't seem to be a problem, e.g., I can type `spack install --cache-only --only dependencies zlib` and it doesn't generate an error.  Furthermore, removing the guard fixes the issue I was showing you in the first stage of specs following the bootstrapping phase (where those pkgs were trying to re-install their bootstrapped compiler from source).

So let's keep this commit!